### PR TITLE
updating typescript packages

### PR DESCRIPTION
**Pull request by luu-alex**
## Description

fix builds for 1.x #5675

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth2-core/package.json [hunk 1]
```diff
@@ -28,6 +28,6 @@
         "@types/node": "^14.14.22",
         "jest": "^26.5.3",
         "ts-jest": "^26.4.4",
-        "typescript": "^3.9.7"
+        "typescript": "^4.9.3"
```
Comment (Spacesai1or) [inline packages/web3-eth2-core/package.json]: Was there a specific reason to bump this to `4.9.3` instead of matching the other packages with `v4.1`?

**Comment by Spacesai1or** [comment]
`web3-eth2-beaconchain`'s `package.json` still has `"typescript": "^4.0.3"`

File: packages/web3-eth2-core/package.json (hunk 1 above)
Comment (luu-alex) [reply packages/web3-eth2-core/package.json]: Missed this thanks

**Review by Spacesai1or — APPROVED** [review]
Thank you Alex for debugging this

**Review by Muhammad-Altabba — APPROVED** [review]
Thanks @luu-alex

File: package.json [hunk 2]
```diff
@@ -133,7 +133,7 @@
         "rimraf": "^3.0.2",
         "sandboxed-module": "^2.0.4",
         "ts-node": "^9.0.0",
-        "typescript": "^3.9.7",
+        "typescript": "4.1",
```
Comment (Muhammad-Altabba) [inline package.json:136]: Just wondering, why not to update to the latest typescript version?

```suggestion
        "typescript": "4.9.3",
```

**Review by nikoulai — APPROVED** [review]

**Review by avkos — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (luu-alex) [reply package.json:136]: One of the reasons was the latest typescript version increases our browser build size by a substantial amount, as well i dont believe theres a reason to upgrade as these changes wont affect us. we only use typescript for the build process on our own type declaration files.

**Pull request merged**